### PR TITLE
Modify StaticResources Cache-Control header, to allow caching large files.

### PR DESCRIPTION
**Pull request by matty256**
To minimize clients downloading large static css/js files.

### CODE REVIEW

File: src/main/scala/bad/robot/temperature/server/StaticResources.scala [hunk 1]
```diff
@@ -20,7 +20,7 @@ object StaticResources {
       resources(request)
 
     response.map {
-      case Successful(resp) => resp.putHeaders(`Cache-Control`(`no-cache`(), `no-store`, `max-age`(Zero)))
+      case Successful(resp) => resp.putHeaders(`Cache-Control`(`public`, `max-age`(Duration(31536000, SECONDS))))
```
Comment (tobyweston) [inline src/main/scala/bad/robot/temperature/server/StaticResources.scala:23]: I might end up changing this to be `365 days` for clarity. The other thing I'm curious about is how clients work out genuine changes to the js or css, is that all handled 'for free'? i.e they will reload if they think they need to?

**Comment by tobyweston** [comment]
Shall we do compression whilst here? Should be [straight forward](http://http4s.org/v0.18/gzip/) but I don't have much experience as to whether its a good idea or not

**Comment by tobyweston** [comment]
This is great, I'm seeing load times reduce form ~10s to < 2s 👍

File: src/main/scala/bad/robot/temperature/server/StaticResources.scala (hunk 1 above)
Comment (matty256) [reply src/main/scala/bad/robot/temperature/server/StaticResources.scala:23]: +1 for changing to seconds to improve readability. Usually a hash or version number is embedded in the static resource name that forces clients to get the new version when it's referenced. During development, it's common to set max-age to zero to force clients to refresh.

**Comment by matty256** [comment]
Compression yes! If the request headers indicate the client supports gzip, we should compress the response, that will make a big difference to decrease initial page load time. Unfortunately I don't know scala/http4s and had no luck finding an example to follow.

**Comment by tobyweston** [comment]
No worries, I can have a go at that. Thanks for the PR!

**Comment by tobyweston** [comment]
Just thinking, shouldn't we use etags here?

**Comment by tobyweston** [comment]
Gzip added in #71

**Comment by matty256** [comment]
Thanks Toby for adding Gzip!

When I worked on web projects 10 years ago, I targeted the simplest
solution for the majority of browsers, i.e. simple cache control headers
for static files. However, today all browsers should correctly support
etags. If it only requires enabling a feature in the http server to
generate and honor etags, I don't see why not. :)

Matt


-- Matt

**Pull request merged**